### ISSUE TITLE
Clarify if `and`, `or`, `xor`, and `not`,  are bitwise or logical

### ISSUE THREAD

**Issue by aabounegm** [objectionary/on-the-origin-of-objects#36]
At the end of section 3 (**Bytes**), there is the following quote:

> There are also attributes for bitwise operations: `★and`, `★or`, `★xor`, `★not`, `★left`, and `★right`.

Which implies that `not` is the bitwise complement operation, and that `and`/`or`/`xor` also operate bitwise.
For example, `not` `00-00` would be `FF-FF`.

However, on the same page, towards the end of subsection 4.2 (**Bool**), there is the following

> The `not` attribute is a Boolean inversion. Attributes `and`, `or`, `xor` are logical conjunction, disjunction, and exclusive disjunction of the object itself with an a array of other Boolean values.

Which implies that these operations are logical (meaning they operate only on `true`/`false` values).
For example, `not` of `00-00` would be `00-01`.

Which of these variants is the correct one?

And as an additional question, what is meant by _"an a array of other Boolean values"_? Do these operations accept an array as an argument or just one other value?

**Comment by yegor256** [comment]
@maxonfjvipon can you please help

**Comment by maxonfjvipon** [comment]
@aabounegm 
1. we don't have `bool` anymore, there are `true` and `false` objects
2. `bytes.and` and `true.and` are different objects. `bytes.and` is bitwise "and", `true.and` is logical "and"

**Comment by aabounegm** [comment]
Yes I remember that blog post. So it's safe to assume that all of those operators (with their respective atoms listed [here](https://github.com/objectionary/normalizer/issues/270)) are bitwise, correct?

**Comment by maxonfjvipon** [comment]
@aabounegm `and`, `or`, `xor`, `not` - yes, bitwise

**Comment by aabounegm** [comment]
Understood, thank you. Feel free to close the issue or keep it open until the paper is updated
